### PR TITLE
Feature/command pattern

### PR DESCRIPTION
**Pull request by t-eliks**

### CODE REVIEW

File: Client/NoOpRunner.Client.Logic/Commands/ChangeVolumeCommand.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        protected override bool PreExecute(int newVolume)
+        {
+            if (settingsViewModel == null)
+                return false;
+
+            if (newVolume < 0)
+                return settingsViewModel?.VolumeLevel - newVolume >= 0;
+
+            return settingsViewModel?.VolumeLevel + newVolume <= 100;
```
Comment (Kodnot) [inline Client/NoOpRunner.Client.Logic/Commands/ChangeVolumeCommand.cs]: Here you increase by `newVolume`, but int execute you just set the volume to `newVolume`? Shouldn't these checks just be 0 <= newVolume <= 100?

File: Core/NoOpRunner.Core/Enums/KeyPress.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    {
+        Up,
+        Right,
+        Left,
+        Space,
+        Down,
+        PowerUp1,
+        PowerUp2
+    }
+}
```
Comment (Kodnot) [inline Core/NoOpRunner.Core/Enums/KeyPress.cs]: Why do you need this?

File: Core/NoOpRunner.Core/NoOpRunner.cs [hunk 3]
```diff
@@ -40,7 +40,7 @@ public Player Player
         }
         public PowerUpsContainer PowerUpsContainer
         {
-            get => GameState.PowerUpsContainer;
+            get => GameState?.PowerUpsContainer;
             set => GameState.PowerUpsContainer = value;
```
Comment (Kodnot) [inline Core/NoOpRunner.Core/NoOpRunner.cs:44]: ? `GameState` only potentially null on the getter? Why?

**Review by Kodnot — CHANGES_REQUESTED** [review]

File: Core/NoOpRunner.Core/Enums/KeyPress.cs (hunk 2 above)
Comment (t-eliks) [reply Core/NoOpRunner.Core/Enums/KeyPress.cs]: On development branch, this file is part of solution but not there physically, had to readd.

File: Core/NoOpRunner.Core/NoOpRunner.cs (hunk 3 above)
Comment (t-eliks) [reply Core/NoOpRunner.Core/NoOpRunner.cs:44]: Had a weird bug with comboboxes.

File: Core/NoOpRunner.Core/Enums/KeyPress.cs (hunk 2 above)
Comment (Kodnot) [reply Core/NoOpRunner.Core/Enums/KeyPress.cs]: Just remove it from the solution instead, I don't think we are using the enum anymore

File: Client/NoOpRunner.Client.Logic/Commands/ChangeVolumeCommand.cs (hunk 1 above)
Comment (t-eliks) [reply Client/NoOpRunner.Client.Logic/Commands/ChangeVolumeCommand.cs]: Fixed

File: Core/NoOpRunner.Core/Enums/KeyPress.cs (hunk 2 above)
Comment (t-eliks) [reply Core/NoOpRunner.Core/Enums/KeyPress.cs]: Removed

**Review by Kodnot — APPROVED** [review]

**Pull request merged**